### PR TITLE
Wizard recipe: yaml_cpp-v0.7.0

### PR DESCRIPTION
**Pull request by pepijndevos**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: yaml_cpp
* Version: v0.7.0

@staticfloat please review and merge.

### CODE REVIEW

File: Y/yaml_cpp/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; ),
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "windows"; )
+]
```
Comment (giordano) [inline Y/yaml_cpp/build_tarballs.jl:35]: Can we build for all platforms and see what happens?
```suggestion
platforms = supported_platforms(; experimental=true)
```

**Comment by pepijndevos** [comment]
It seems the musl targets fail to compile so the wizard removed them. Anything that can be done?
Also added the split abi strings thing as suggested by the warning.

File: Y/yaml_cpp/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+sources = [
+    ArchiveSource("https://github.com/jbeder/yaml-cpp/archive/refs/tags/yaml-cpp-0.7.0.zip", "4d5e664a7fb2d7445fc548cc8c0e1aa7b1a496540eb382d137e2cc263e6d3ef5")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd yaml-cpp-yaml-cpp-0.7.0/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DYAML_BUILD_SHARED_LIBS=ON
+make
```
Comment (giordano) [inline Y/yaml_cpp/build_tarballs.jl:18]: Minor reorganisation
```suggestion
cd $WORKSPACE/srcdir/yaml-cpp-yaml-cpp*/
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DYAML_BUILD_SHARED_LIBS=ON \
    ..
make -j${nproc}
```

**Comment by pepijndevos** [comment]
```
Several platforms failed to build:
 - Linux i686 {libc=musl}
 - Linux x86_64 {libc=musl}
```
Error:
```
Building x86_64-linux-musl:                  from /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/4.8.5/algorithm:62,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:1:
/opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/4.8.5/include/mm_malloc.h:34:64: error: declaration of ‘int posix_memalign(void**, size_t, size_t) throw ()’ has a different exception specifier
 extern "C" int posix_memalign (void **, size_t, size_t) throw ();
                                                                ^
In file included from /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/4.8.5/cstdlib:72:0,
                 from /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/4.8.5/bits/stl_algo.h:59,
                 from /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/4.8.5/algorithm:62,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:1:
/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/stdlib.h:99:5: error: from previous declaration ‘int posix_memalign(void**, size_t, size_t)’
 int posix_memalign (void **, size_t, size_t);
     ^
In file included from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_data.h:17:0,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:13,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_iterator.h: In instantiation of ‘struct YAML::detail::node_iterator_value<YAML::detail::node>’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/iterator.h:80:10:   required from here
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_iterator.h:26:8: warning: base class ‘struct std::pair<YAML::detail::node*, YAML::detail::node*>’ has a non-virtual destructor [-Weffc++]
 struct node_iterator_value : public std::pair<V*, V*> {
        ^
In file included from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_data.h:18:0,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:13,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/iterator.h:20:8: warning: base class ‘class YAML::Node’ has a non-virtual destructor [-Weffc++]
 struct iterator_value : public Node, std::pair<Node, Node> {
        ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/iterator.h:20:8: warning: base class ‘struct std::pair<YAML::Node, YAML::Node>’ has a non-virtual destructor [-Weffc++]
In file included from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:12:0,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h: In member function ‘void YAML::detail::node_ref::set_mark(const YAML::Mark&)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:33:35: warning: declaration of ‘mark’ shadows a member of 'this' [-Wshadow]
   void set_mark(const Mark& mark) { m_pData->set_mark(mark); }
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h: In member function ‘void YAML::detail::node_ref::set_type(YAML::NodeType::value)’:
   void set_type(NodeType::value type) { m_pData->set_type(type); }
                                       ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h: In member function ‘void YAML::detail::node_ref::set_tag(const string&)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:35:40: warning: declaration of ‘tag’ shadows a member of 'this' [-Wshadow]
   void set_tag(const std::string& tag) { m_pData->set_tag(tag); }
                                        ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h: In member function ‘void YAML::detail::node_ref::set_scalar(const string&)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:37:46: warning: declaration of ‘scalar’ shadows a member of 'this' [-Wshadow]
                                              ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_ref.h:38:45: warning: declaration of ‘style’ shadows a member of 'this' [-Wshadow]
   void set_style(EmitterStyle::value style) { m_pData->set_style(style); }
                                             ^
In file included from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:12:0,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h: In member function ‘void YAML::detail::node::set_mark(const YAML::Mark&)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:74:35: warning: declaration of ‘mark’ shadows a member of 'this' [-Wshadow]
                                   ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:76:39: warning: declaration of ‘type’ shadows a member of 'this' [-Wshadow]
                                       ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h: In member function ‘void YAML::detail::node::set_scalar(const string&)’:
   void set_scalar(const std::string& scalar) {
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h: In member function ‘void YAML::detail::node::set_tag(const string&)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:89:40: warning: declaration of ‘tag’ shadows a member of 'this' [-Wshadow]
   void set_tag(const std::string& tag) {
                                        ^
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h: In member function ‘void YAML::detail::node::set_style(YAML::EmitterStyle::value)’:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:95:45: warning: declaration of ‘style’ shadows a member of 'this' [-Wshadow]
   void set_style(EmitterStyle::value style) {
                                             ^
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node_data.h:18,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/detail/node.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:12,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/node.h: At global scope:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/node.h:29:20: warning: unused parameter ‘rhs’ [-Wunused-parameter]
 class YAML_CPP_API Node {
                    ^
In file included from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/convert.h:20:0,
                 from /workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/src/convert.cpp:3:
/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0/include/yaml-cpp/node/impl.h:45:38: note: synthesized method ‘YAML::Node::Node(const YAML::Node&)’ first required here 
 inline Node::Node(const Node& rhs) = default;
                                      ^
make[2]: *** [CMakeFiles/yaml-cpp.dir/build.make:122: CMakeFiles/yaml-cpp.dir/src/convert.cpp.o] Error 1
make[2]: Leaving directory '/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0'
make[1]: *** [CMakeFiles/Makefile2:1015: CMakeFiles/yaml-cpp.dir/all] Error 2
make[1]: Leaving directory '/workspace/srcdir/yaml-cpp-yaml-cpp-0.7.0'
make: *** [Makefile:161: all] Error 2

```

File: Y/yaml_cpp/build_tarballs.jl (hunk 1 above)
Comment (pepijndevos) [reply Y/yaml_cpp/build_tarballs.jl:35]: As I wrote above, x86 with musl fails

File: Y/yaml_cpp/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+products = [
+    LibraryProduct("libyaml-cpp", :libyaml)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline Y/yaml_cpp/build_tarballs.jl:53]: Oh, for those Musl issues we need to require GCC 6:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"6")
```

**Pull request merged**